flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/FlinkFilterJoinRule.java

```diff
@@ -128,7 +129,9 @@ protected void perform(RelOptRuleCall call, Filter filter, Join join) {
                 joinType,
                 true,
                 !joinType.generatesNullsOnLeft(),
-                !joinType.generatesNullsOnRight(),
+                !joinType.generatesNullsOnRight()
+                        && !TemporalJoinUtil.containsInitialTemporalJoinCondition(
```

Comment:
the filter added here and line 176 below are not enough to prevent pushing down a filter into the right   side input of event time temporal join, because this rule will also perform pushing both above and join filters to opposite side by transitive rule, e.g., think of such an query:

```sql
SELECT *      
FROM Orders AS o JOIN
rates_last_row_rowtime FOR SYSTEM_TIME AS OF o.rowtime AS r
ON o.currency = r.currency AND o.currency = 10
```
though the local predicate 'o.currency = 10' is releated to left side, but by this transitive optimization, it will push a new filter 'r.currency = 10' to the right side, this will break our target, so we need to figure out a way to cover this case

Reply:
@lincoln-lil If there was a constant equi condition on join key, the `JoinReduceExpressionsRule` in calcite will try to reduce all InputRef in JoinCondition to be constant expression, while generating call expression for __TEMPORAL_JOIN_LEFT_KEY(INT NOT NULL) is not allowed. Maybe we should make the __TEMPORAL_JOIN_LEFT_KEY,  __TEMPORAL_JOIN_LEFT_KEY and __INITIAL_TEMPORAL_JOIN_CONDITION irreducible.

Reply:
@shuiqiangchen If the `JoinReduceExpressionsRule` does an unintended constant reducing of temporal's initial conditions, we certainly need to extend and modify it (e.g., add a new FlinkJoinReduceExpressionsRule). But what may be bigger is another problem, you can refer to the previous message, @godfreyhe and I have not been satisfied with the solution after offline discussion.

Reply:
I think we can only deal with the event time temporal join since processing time temporal join is unsupported for now.